src/main/java/wolox/bootstrap/security/AuthenticationFilter.java

```diff
@@ -0,0 +1,64 @@
+package wolox.bootstrap.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import wolox.bootstrap.models.ApplicationUser;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.security.Key;
+import java.util.ArrayList;
+import java.util.Date;
+
+
+import static wolox.bootstrap.constants.SecurityConstants.EXPIRATION_TIME;
+import static wolox.bootstrap.constants.SecurityConstants.KEY;
+
+public class AuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+    private AuthenticationManager authenticationManager;
+
+    public AuthenticationFilter(AuthenticationManager authenticationManager) {
+        this.authenticationManager = authenticationManager;
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest req,
+                                                HttpServletResponse res) throws AuthenticationException {
+        try {
+            ApplicationUser applicationUser = new ObjectMapper().readValue(req.getInputStream(), ApplicationUser.class);
+            return authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(applicationUser.getUsername(),
+                            applicationUser.getPassword(), new ArrayList<>())
+            );
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected void successfulAuthentication(HttpServletRequest req, HttpServletResponse res, FilterChain chain,
```

Comment:
JWT token usually contains a hash that is saved in the ApplicationUser table. That way you can compare them in the other endpoints and make all tokens invalid by changing the hash in the table.
I recommend you to try to implement that, i know it´s not easy.